ethereum/blockwatch/block_watcher_test.go

```diff
@@ -67,25 +73,410 @@ func TestWatcher(t *testing.T) {
+				"10-20": filterLogsResponse{
+					Logs: []types.Log{
+						logStub,
+					},
+				},
+			},
+			Logs: []types.Log{logStub},
+		},
+		filterLogsRecusivelyTestCase{
+			Label: "TOO_MANY_RESULTS_INFURA_ERROR",
+			rangeToFilterLogsResponse: map[string]filterLogsResponse{
+				"10-20": filterLogsResponse{
+					Err: errors.New(infuraTooManyResultsErrMsg),
+				},
+				"10-15": filterLogsResponse{
+					Logs: []types.Log{
+						logStub,
+					},
+				},
+				"16-20": filterLogsResponse{
+					Logs: []types.Log{
+						logStub,
+					},
+				},
+			},
+			Logs: []types.Log{logStub, logStub},
+		},
+		filterLogsRecusivelyTestCase{
+			Label: "TOO_MANY_RESULTS_INFURA_ERROR_DEEPER_RECURSION",
+			rangeToFilterLogsResponse: map[string]filterLogsResponse{
+				"10-20": filterLogsResponse{
+					Err: errors.New(infuraTooManyResultsErrMsg),
+				},
+				"10-15": filterLogsResponse{
+					Logs: []types.Log{
+						logStub,
+					},
+				},
+				"16-20": filterLogsResponse{
+					Err: errors.New(infuraTooManyResultsErrMsg),
+				},
+				"16-18": filterLogsResponse{
+					Logs: []types.Log{
+						logStub,
+					},
+				},
+				"19-20": filterLogsResponse{
+					Logs: []types.Log{
+						logStub,
+					},
+				},
+			},
+			Logs: []types.Log{logStub, logStub, logStub},
+		},
+		filterLogsRecusivelyTestCase{
+			Label: "TOO_MANY_RESULTS_INFURA_ERROR_DEEPER_RECURSION_FAILURE",
+			rangeToFilterLogsResponse: map[string]filterLogsResponse{
+				"10-20": filterLogsResponse{
+					Err: errors.New(infuraTooManyResultsErrMsg),
+				},
+				"10-15": filterLogsResponse{
+					Logs: []types.Log{
+						logStub,
+					},
+				},
+				"16-20": filterLogsResponse{
+					Err: errUnexpected,
+				},
+			},
+			Err: errUnexpected,
+		},
+		filterLogsRecusivelyTestCase{
+			Label: "UNEXPECTED_ERROR",
+			rangeToFilterLogsResponse: map[string]filterLogsResponse{
+				"10-20": filterLogsResponse{
+					Err: errUnexpected,
+				},
+			},
+			Err: errUnexpected,
+		},
+	}
+
+	meshDB, err := meshdb.NewMeshDB("/tmp/leveldb_testing/" + uuid.New().String())
+	require.NoError(t, err)
+	config.MeshDB = meshDB
+
+	for _, testCase := range testCases {
+		fakeLogClient, err := newFakeLogClient(testCase.rangeToFilterLogsResponse)
+		require.NoError(t, err)
+		config.Client = fakeLogClient
+		watcher := New(config)
+
+		logs, err := watcher.filterLogsRecurisively(from, to, []types.Log{})
+		require.Equal(t, testCase.Err, err, testCase.Label)
+		require.Equal(t, testCase.Logs, logs, testCase.Label)
+		assert.Equal(t, len(testCase.rangeToFilterLogsResponse), fakeLogClient.Count())
+	}
+}
+
+type logsInBlockRangeTestCase struct {
+	Label                     string
+	From                      int
+	To                        int
+	RangeToFilterLogsResponse map[string]filterLogsResponse
+	Logs                      []types.Log
+	FurthestBlockProcessed    int
+}
+
+func TestGetLogsInBlockRange(t *testing.T) {
+	from := 10
+	to := 20
+	maxBlocksInQuery := maxBlocksInGetLogsQuery
```

Comment:
It looks like we don't need this variable and can simply refer to `maxBlocksInGetLogsQuery` directly. (Previously it involved a cast to `int` but that has since been removed).